src/swupd/manifest.go

```diff
@@ -226,3 +226,111 @@ func (m *Manifest) ReadManifestFromFile(f string) error {
 	// return err so the deferred close can modify it
 	return err
 }
+
+// getManifestTimeStamp returns the unix timestamp of m.Header.TimeStamp
+func (m *Manifest) getManifestTimeStamp() int64 {
+	return m.Header.TimeStamp.Unix()
+}
+
+// writeManifestFileHeader writes the header of a manifest to the file
+func writeManifestFileHeader(m *Manifest, w *bufio.Writer) error {
+	var err error
+	if err = m.CheckHeaderPopulated(); err != nil {
+		return err
+	}
+
+	if _, err = w.WriteString(fmt.Sprintf("MANIFEST\t%v\n", m.Header.Format)); err != nil {
+		return err
+	}
+	if _, err = w.WriteString(fmt.Sprintf("version:\t%v\n", m.Header.Version)); err != nil {
+		return err
+	}
+	if _, err = w.WriteString(fmt.Sprintf("previous:\t%v\n", m.Header.Previous)); err != nil {
+		return err
+	}
+	if _, err = w.WriteString(fmt.Sprintf("filecount:\t%v\n", m.Header.FileCount)); err != nil {
+		return err
+	}
+	if _, err = w.WriteString(fmt.Sprintf("timestamp:\t%v\n", m.getManifestTimeStamp())); err != nil {
+		return err
+	}
+	if _, err = w.WriteString(fmt.Sprintf("contentsize:\t%v\n", m.Header.ContentSize)); err != nil {
+		return err
+	}
+
+	for _, include := range m.Header.Includes {
+		if _, err = w.WriteString(fmt.Sprintf("includes:\t%v\n", include.Name)); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// writeManifestFileSep writes a blank line to the file
+func writeManifestFileSep(w *bufio.Writer) error {
+	_, err := w.WriteString("\n")
+	return err
+}
+
+// writeManifestFileEntry writes the file entry line to the file
+func writeManifestFileEntry(file *File, w *bufio.Writer) error {
+	var flags string
+	var err error
+	if flags, err = file.getFlagString(); err != nil {
+		return err
+	}
+
+	line := fmt.Sprintf("%v\t%v\t%v\t%v\n",
+		flags,
+		file.getHashString(),
+		file.Version,
+		file.Name)
+	_, err = w.WriteString(line)
+	return err
+}
+
+// writeManifestFileEntries writes all the entries in files to the file
+func writeManifestFileEntries(m *Manifest, w *bufio.Writer) error {
+	for _, entry := range m.Files {
+		if err := writeManifestFileEntry(entry, w); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// WriteManifestFile writes manifest m to a new file at path
+func (m *Manifest) WriteManifestFile(path string) error {
+	var err error
+	var f *os.File
+	if f, err = os.Create(path); err != nil {
+		return err
+	}
+
+	// handle close errors
+	defer func() {
+		cerr := f.Close()
+		if err == nil {
+			err = cerr
+		}
+	}()
+
+	w := bufio.NewWriter(f)
+
+	if err = writeManifestFileHeader(m, w); err != nil {
+		return err
+	}
+
+	if err = writeManifestFileSep(w); err != nil {
+		return err
+	}
+
+	if err = writeManifestFileEntries(m, w); err != nil {
+		return err
+	}
+
+	// return error code or nil from w.Flush()
```

Comment:
Does the error from w.Flush() get overwritten if there is an error in the defer function?

Reply:
oh perhaps. Maybe I should do:
```
err = w.Flush()
return err
```

Reply:
done